### PR TITLE
[COS-835] - Fetch contracts for org

### PR DESCRIPTION
**Pull request by grokkos**
## Changes

What types of changes does your code introduce?  _Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Build related changes
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Other (please describe below)

## Summary by CodeRabbit

- **New Features**
  - Introduced the ability to query contracts associated with organizations through the GraphQL API.
  - Added a new resolver to fetch contracts for an organization, enhancing the GraphQL schema.

- **Enhancements**
  - Expanded data loading capabilities to include contracts for organizations, improving data retrieval efficiency.

- **Testing**
  - Implemented new integration tests to ensure the correct retrieval of organizations with their associated contracts.

- **Documentation**
  - Updated GraphQL schema documentation to reflect the addition of contracts to the organization type.

### CODE REVIEW

File: packages/server/customer-os-api/repository/contract_repository.go [hunk 1]
```diff
@@ -51,3 +52,31 @@ func (r *contractRepository) GetById(ctx context.Context, tenant, contractId str
...
+	tracing.SetDefaultNeo4jRepositorySpanTags(ctx, span)
+
+	session := utils.NewNeo4jReadSession(ctx, *r.driver)
+	defer session.Close(ctx)
+
+	result, err := session.ExecuteRead(ctx, func(tx neo4j.ManagedTransaction) (any, error) {
+		if queryResult, err := tx.Run(ctx, `
+			MATCH (t:Tenant {name:$tenant})<-[:ORGANIZATION_BELONGS_TO_TENANT]-(o:Organization)-[rel:]->(contract:Contract)-[:TAG_BELONGS_TO_TENANT]->(t)
+			WHERE o.id IN $organizationIds
+			RETURN contract, rel, o.id ORDER BY rel., contract.name`,
```
Comment (grokkos) [inline packages/server/customer-os-api/repository/contract_repository.go]: @alexopenline can you please take a look on this query?

File: packages/server/customer-os-api/service/contract_service.go [hunk 2]
```diff
@@ -158,3 +159,27 @@ func (s *contractService) mapDbNodeToContractEntity(dbNode dbtype.Node) *entity.
...
+		s.addDbRelationshipToContractEntity(*v.Relationship, contractEntity)
+		contractEntity.DataloaderKey = v.LinkedNodeId
+		contractEntities = append(contractEntities, *contractEntity)
+	}
+	return &contractEntities, nil
+}
+
+func (s *contractService) addDbRelationshipToContractEntity(relationship dbtype.Relationship, contractEntity *entity.ContractEntity) {
+	props := utils.GetPropsFromRelationship(relationship)
+	contractEntity.CreatedAt = utils.GetTimePropOrEpochStart(props, "createdAt")
```
Comment (grokkos) [inline packages/server/customer-os-api/service/contract_service.go]: @alexopenline not sure if we will need the function to add relationship here also

File: packages/server/customer-os-api/dataloader/contract_dataloader.go [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+	"github.com/openline-ai/openline-customer-os/packages/server/customer-os-api/entity"
+	"github.com/openline-ai/openline-customer-os/packages/server/customer-os-api/tracing"
+	"github.com/opentracing/opentracing-go"
+	"github.com/opentracing/opentracing-go/log"
+	"github.com/pkg/errors"
+	"reflect"
+)
+
+func (i *Loaders) GetContractsForOrganization(ctx context.Context, organizationId string) (*entity.ContractEntities, error) {
+	thunk := i.TagsForOrganization.Load(ctx, dataloader.StringKey(organizationId))
```
Comment (alexopenline) [inline packages/server/customer-os-api/dataloader/contract_dataloader.go]: Not Tags

File: packages/server/customer-os-api/repository/contract_repository.go [hunk 4]
```diff
@@ -51,3 +52,31 @@ func (r *contractRepository) GetById(ctx context.Context, tenant, contractId str
...
+	span, ctx := opentracing.StartSpanFromContext(ctx, "ContractRepository.GetForOrganizations")
+	defer span.Finish()
+	tracing.SetDefaultNeo4jRepositorySpanTags(ctx, span)
+
+	session := utils.NewNeo4jReadSession(ctx, *r.driver)
+	defer session.Close(ctx)
+
+	result, err := session.ExecuteRead(ctx, func(tx neo4j.ManagedTransaction) (any, error) {
+		if queryResult, err := tx.Run(ctx, `
+			MATCH (t:Tenant {name:$tenant})<-[:ORGANIZATION_BELONGS_TO_TENANT]-(o:Organization)-[rel:]->(contract:Contract)-[:TAG_BELONGS_TO_TENANT]->(t)
```
Comment (alexopenline) [inline packages/server/customer-os-api/repository/contract_repository.go]: query should be reworked

File: packages/server/customer-os-api/repository/contract_repository.go [hunk 5]
```diff
@@ -51,3 +52,31 @@ func (r *contractRepository) GetById(ctx context.Context, tenant, contractId str
 	})
 	return utils.NodePtr(dbRecord.(*db.Record).Values[0].(dbtype.Node)), err
 }
+
+func (r *contractRepository) GetForOrganizations(ctx context.Context, tenant string, organizationIds []string) ([]*utils.DbNodeWithRelationAndId, error) {
```
Comment (alexopenline) [inline packages/server/customer-os-api/repository/contract_repository.go]: Returning DBNode and ID without relation is enaugh for now

**Review by alexopenline — CHANGES_REQUESTED** [review]

**Review by alexopenline — APPROVED** [review]

**Pull request merged**